external-applications/quarkus-workshop-super-heroes/src/test/java/io/quarkus/ts/openshift/heroes/workshop/HeroesOpenShiftIT.java

```diff
@@ -0,0 +1,175 @@
+                DEFAULT_POWERS);
+
+        String location = given()
+                .body(rb)
+                .header(CONTENT_TYPE, APPLICATION_JSON)
+                .header(ACCEPT, APPLICATION_JSON)
+                .when()
+                .post(url + "/api/heroes")
+                .then()
+                .statusCode(CREATED.getStatusCode())
+                .extract().header("Location");
+        assertTrue(location.contains("/api/heroes"));
+
+
+        String[] segments = location.split("/");
+        heroId = segments[segments.length - 1];
+        assertNotNull(heroId);
+
+        given()
+                .pathParam("id", heroId)
+                .when().get("/api/heroes/{id}")
+                .then()
+                .statusCode(OK.getStatusCode())
+                .header(CONTENT_TYPE, APPLICATION_JSON)
+                .body("name", Is.is(DEFAULT_NAME))
+                .body("otherName", Is.is(DEFAULT_OTHER_NAME))
+                .body("level", Is.is(DEFAULT_LEVEL*3))
+                .body("picture", Is.is(DEFAULT_PICTURE))
+                .body("powers", Is.is(DEFAULT_POWERS));
+    }
+
+    @Test
+    @Order(2)
+    public void testUpdateHero() {
+        String rb = getHeroJson(heroId, UPDATED_NAME, UPDATED_OTHER_NAME, Integer.toString(UPDATED_LEVEL),
+                UPDATED_PICTURE, UPDATED_POWERS);
+
+        given()
+                .body(rb)
+                .header(CONTENT_TYPE, APPLICATION_JSON)
+                .header(ACCEPT, APPLICATION_JSON)
+                .when()
+                .put("/api/heroes")
+                .then()
+                .statusCode(OK.getStatusCode())
+                .header(CONTENT_TYPE, APPLICATION_JSON)
+                .body("name", Is.is(UPDATED_NAME))
+                .body("otherName", Is.is(UPDATED_OTHER_NAME))
+                .body("level", Is.is(UPDATED_LEVEL))
+                .body("picture", Is.is(UPDATED_PICTURE))
+                .body("powers", Is.is(UPDATED_POWERS));
+    }
+
+    @Test
+    @Order(3)
+    public void testDeleteHero() {
+        given()
+                .pathParam("id", heroId)
+                .when().delete("/api/heroes/{id}")
+                .then()
+                .statusCode(NO_CONTENT.getStatusCode());
+    }
+
+    private static String getHeroJson(String id, String name, String otherName, String level, String picture,
+                                      String powers) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("{");
+        if (id != null && !id.isEmpty()) { sb.append("\"id\":" + id + ","); }
+        sb.append("\"name\":\"" + name + "\"");
+        sb.append(", \"otherName\":\"" + otherName + "\"");
+        sb.append(", \"level\":" + level);
+        sb.append(", \"picture\":\"" + picture + "\"");
+        sb.append(", \"powers\":\"" + powers + "\"");
+        sb.append("}");
+        return sb.toString();
```

Comment:
Use Jackson or JSON-P?

Reply:
an example can be found here - https://javaee.github.io/jsonp/getting-started.html

Reply:
Actually, seems like [body](https://github.com/rest-assured/rest-assured/blob/master/rest-assured/src/main/java/io/restassured/specification/RequestSpecification.java#L151) method in RestAssured takes care of serialization, so I decided to use that.

Reply:
That's even better, for some reason I forgot about it :-)